app/models/container_label_tag_mapping.rb

```diff
@@ -2,15 +2,99 @@ class ContainerLabelTagMapping < ApplicationRecord
   # A mapping matches labels on `resource_type` (NULL means any), `name` (required),
   # and `value` (NULL means any).
   #
-  # Different labels might map to one tag, and one label might map to multiple tags.
+  # Different labels might map to same tag, and one label might map to multiple tags.
   #
   # There are 2 kinds of rows:
   # - When `label_value` is specified, we map only this value to a specific `tag`.
   # - When `label_value` is NULL, we map this name with any value to per-value tags.
   #   In this case, `tag` specifies the category under which to create
   #   the value-specific tag (and classification) on demand.
   #   We then also add a specific `label_value`->specific `tag` mapping here.
+
   belongs_to :tag
 
-  validates :label_name, :presence => true
+  def self.drop_cache
+    @cached_hash = nil
+    @cached_mapped_tags = nil
```

Comment:
@kbrock could we get your opinion on the caches above? 
Is there maybe a more streamlined way of doing that?